Fix redirect getting overwritten in localStorage

JerrettDavis opened the pull request:
Issue #1060 refers to application's having their own internal `redirect` record in `localStorage` overwritten by the `AutoLoginService`. The `StoragePersistenceService` already handles prefixing stored values to remedy this issue. I've converted the `AutoLoginService` to use `StoragePersistenceService` instead of `localStorage`.

FabianGosebrink commented:
You are on fire! Thanks :-) We will look into this, a bit short of time currently, and then merge this. Thanks for your help, we really appreciate it!

JerrettDavis commented:
We've been using the library for a bit now, so I'm always happy to lend a hand. On that note, I still need to fix the unit tests before this can be merged. I should have that done shortly.

JerrettDavis commented:
I've updated all the necessary unit tests to make them compatible with the change away from `localStorage`. One minor change I implemented throughout was the inclusion of a `clear` method on the `AbstractSecurityStorage` class. This shouldn't be at all breaking, but it enables `localStorage`-based Unit Tests to perform post-test cleanup, if necessary. As it stands now, it shouldn't be necessary as the mock used for testing doesn't actually persist anything, but previously it was and I didn't want any sort of feature-regression.

FabianGosebrink commented:
Glad to hear that you are using the lib :-) Thanks for being part of this then :) We have to test the AutoLogin Good and the Popup. @damienbod and I will run the tests and then this will land in V12. Thanks!
